### PR TITLE
implement To/FromVariant for OS strings and paths

### PR DESCRIPTION
**Pull request by euclio**
The impls use the platform-specific byte representation for serialization, so they're `cfg`'d for `unix` and `windows`.

### CODE REVIEW

**Review by sdroege — COMMENTED** [review]
This seems a bit suboptimal because the variant type is different between platforms, so serializations created on e.g. Linux won't be possible to open on Windows. Not sure if that's acceptable.

@bilelmoussaoui?

**Comment by bilelmoussaoui** [comment]
> This seems a bit suboptimal because the variant type is different between platforms, so serializations created on e.g. Linux won't be possible to open on Windows. Not sure if that's acceptable.
> 
> @bilelmoussaoui?

I think it is okay to merge this if it is properly documented in https://gtk-rs.org/gtk-rs-core/git/docs/glib/variant/index.html

**Comment by sdroege** [comment]
> I think it is okay to merge this if it is properly documented in https://gtk-rs.org/gtk-rs-core/git/docs/glib/variant/index.html

Where would you document it? Both on the `impl` and at the top somewhere?

**Comment by bilelmoussaoui** [comment]
> > I think it is okay to merge this if it is properly documented in https://gtk-rs.org/gtk-rs-core/git/docs/glib/variant/index.html
> 
> Where would you document it? Both on the `impl` and at the top somewhere?

Something like that yes, also maybe add an example on the page i mentioned above

**Comment by euclio** [comment]
Pushed a new revision with the requested documentation updates.

**Review by bilelmoussaoui — APPROVED** [review]

File: glib/src/variant.rs [hunk 1]
```diff
@@ -9,9 +9,18 @@
 //! `Variant` types are described by [`VariantType`](../struct.VariantType.html)
 //! "type strings".
 //!
-//! Although `GVariant` supports arbitrarily complex types, this binding is
-//! currently limited to the basic ones: `bool`, `u8`, `i16`, `u16`, `i32`,
-//! `u32`, `i64`, `u64`, `f64`, `&str`/`String`, and [`VariantDict`](../struct.VariantDict.html).
+//! Although `GVariant` supports arbitrarily complex types, this crate only provides bindings for a
```
Comment (sdroege) [inline glib/src/variant.rs:12]: The caveat at the beginning is not really true anymore. We support arbitrarily complex types in the bindings now.

**Review by sdroege — COMMENTED** [review]
Looks good to me otherwise

File: glib/src/variant.rs [hunk 2]
```diff
@@ -9,9 +9,18 @@
...
+//! subset of common types. This subset includes primitives, strings, data structures from the
+//! standard library, and [`VariantDict`](crate::VariantDict). See [`ToVariant#foreign-impls`] for
```
Comment (sdroege) [inline glib/src/variant.rs:14]:
```suggestion
//! `GVariant` supports arbitrarily complex types built from primitives like integers, floating point
//! numbers, strings, arrays, tuples and dictionaries. See [`ToVariant#foreign-impls`] for
```

**Comment by sdroege** [comment]
I'll backport this to 0.15.3 some time in the next days.

**Pull request merged**